cmd/subctl/diagnose.go

```diff
@@ -150,6 +150,34 @@ var (
 		},
 	}
 
+	diagnoseFirewallNatDiscovery = &cobra.Command{
+		Use:   "nat-discovery <localkubeconfig> <remotekubeconfig>",
+		Short: "Check firewall access for nat-discovery to function properly",
+		Long:  "This command checks if the firewall configuration allows nat-discovery between the configured Gateway nodes.",
+		Args: func(command *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				return fmt.Errorf("two kubeconfigs must be specified")
+			}
+
+			same, err := compareFiles(args[0], args[1])
+			if err != nil {
+				return err
+			}
+
+			if same {
+				return fmt.Errorf("the specified kubeconfig files are the same")
+			}
+
+			return nil
+		},
```

Comment:
This block is duplicated with L130-143. Can we move it to a function:

```Go
func checkKubeconfigArgs(_ *cobra.Command, args []string) error {
   ....
}
```

```suggestion
		Args: checkKubeconfigArgs,
```